clippy_lints/src/manual_unwrap_or_default.rs

```diff
@@ -131,6 +131,17 @@ fn handle<'tcx>(cx: &LateContext<'tcx>, if_let_or_match: IfLetOrMatch<'tcx>, exp
         },
     };
 
+    // // We check if the expression is not a None variant
+    if let Some(none_def_id) = cx.tcx.lang_items().option_none_variant() {
+        if let ExprKind::Path(QPath::Resolved(_, path)) = &condition.kind {
+            if let Some(def_id) = path.res.opt_def_id() {
+                if cx.tcx.parent(def_id) == none_def_id {
+                    return;
+                }
+            }
+        }
+    }
```

Comment:
This might be equivalent to: 
```rust
if clippy_utils::is_path_lang_item(cx, condition, LangItem::OptionNone) {
    return;
}
```

also, would you mind add some test cases to prove that this could fix the issue? :thinking:

Reply:
Okay, thanks ! 

Yes, sure I will look how to add a test

Reply:
![image](https://github.com/rust-lang/rust-clippy/assets/148005618/defb989d-1dc6-4256-a4a3-82fc074ad9e5)

Your solution doesn't seem to work

Reply:
Oops, sorry my bad, somehow I missed the `cx.tcx.parent(def_id)`

Reply:
> Yes, sure I will look how to add a test

You can put test codes in `tests/ui/manual_unwrap_or_default.rs`, then run `TESTNAME='manual_unwrap_or_default' cargo uitest`

the details are in [the clippy book](https://doc.rust-lang.org/nightly/clippy/development/writing_tests.html)

Reply:
The tests are there already